platforms/access-bridge/src/errors.ts

```diff
@@ -0,0 +1,205 @@
+  [ErrorCode.NotFound]: 'The requested resource could not be found.',
+  [ErrorCode.MethodNotAllowed]: 'The requested resource only supports {value} requests.',
+  [ErrorCode.InternalError]: 'An error was encountered while processing the request. Please try again.',
+};
+
+/**
+ * Allowed response status codes.
+ */
+export type ErrorStatusCode = 400 | 401 | 403 | 404 | 405 | 500;
+
+/**
+ * Base class for errors.
+ */
+export abstract class AccessBridgeError {
+  private readonly descriptionOverride: string | null;
+
+  protected constructor(description?: string) {
+    this.descriptionOverride = description || null;
+  }
+
+  abstract get code(): ErrorCode;
+  abstract get statusCode(): ErrorStatusCode;
+  abstract get headers(): Iterable<[string, string]>;
+  protected abstract get defaultDescription(): string;
+
+  get description(): string {
+    return this.descriptionOverride || this.defaultDescription;
+  }
+}
+
+/**
+ * Factory for generating error classes.
+ * @param errorCode - The error code for errors of this type.
+ * @param statusCode - The corresponding HTTP status code.
+ * @param defaultDescriptionFactory - A function to fill in the default description based on the error context.
+ * @param headersFactory - A function to fill in the error response headers based on the error context.
+ * @returns a new error class.
+ */
+function createError<T>(
+  errorCode: ErrorCode,
+  statusCode: ErrorStatusCode,
+  defaultDescriptionFactory: (context: T) => string,
+  headersFactory: (context: T) => Iterable<[string, string]> = () => []
+): new (context: T & { description?: string }) => AccessBridgeError {
+  return class extends AccessBridgeError {
+    private readonly context: T;
+
+    constructor(context: T & { description?: string }) {
+      super(context?.description);
+      this.context = context;
+    }
+
+    get code(): ErrorCode {
+      return errorCode;
+    }
+
+    get statusCode(): ErrorStatusCode {
+      return statusCode;
+    }
+
+    get headers(): Iterable<[string, string]> {
+      return headersFactory(this.context);
+    }
+
+    protected get defaultDescription(): string {
+      return defaultDescriptionFactory(this.context);
+    }
+  };
+}
+
+/**
+ * Retrieves the description for a given error code.
+ *
+ * This function replaces placeholders in the error description with the provided `value` and `reason`.
+ *
+ * @param {ErrorCode} code - The error code for which the description is needed.
+ * @param {string} [value] - An optional value to replace the `{value}` placeholder in the description.
+ * @param {string} [reason] - An optional reason to replace the `{reason}` placeholder in the description.
+ * @returns {string} The description corresponding to the error code, with the `{value}` and `{reason}`
+ * placeholders replaced by the provided arguments if applicable.
+ */
+export function getErrorCodeDescription(code: ErrorCode, value?: string, reason?: string): string {
```

Comment:
There seems to be a lot of boilerplate repeat of getErrorCodeDescription. Can you put it in the abstract class?